core/sail/memory/src/main/java/org/eclipse/rdf4j/sail/memory/MemorySailStore.java

```diff
@@ -427,16 +427,18 @@ public synchronized void prepare() throws SailException {
 
 		@Override
 		public synchronized void flush() throws SailException {
+
+		}
+
+		@Override
+		public void close() {
 			if (txnLock) {
 				currentSnapshot = Math.max(currentSnapshot, nextSnapshot);
 				if (requireCleanup) {
 					scheduleSnapshotCleanup();
 				}
 			}
```

Comment:
Moved the transaction incrementation into the closing of the branch instead of at the flushing or the branch.

Reply:
I think that makes sense. Part of the problem with with this entire structure is that none of this is really documented - I keep finding myself trying to puzzle out the intent.

Reply:
I think that makes sense. Part of the problem with with this entire structure is that none of this is really documented - I keep finding myself trying to puzzle out the intent.